kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerControllerImpl.java

```diff
@@ -209,22 +220,26 @@ public synchronized KieServerSetup connect(KieServerInfo serverInfo) {
 
     @Override
     public synchronized void disconnect(KieServerInfo serverInfo) {
-        ServerTemplate serverTemplate = templateStorage.load(serverInfo.getServerId());
+        String id = serverInfo.getServerId();
+        ServerTemplate serverTemplate = templateStorage.load(id);
+        ServerInstanceKey serverInstanceKey = new ServerInstanceKey(id, id, id, "");
         if (serverTemplate != null) {
             logger.debug("Server id {} known to the controller, checking if given server exists", serverInfo.getServerId());
-
-            if (serverTemplate != null) {
-                logger.info("Server {} disconnected from controller", serverInfo.getLocation());
-                ServerInstanceKey serverInstanceKey = ModelFactory.newServerInstanceKey(serverInfo.getServerId(), serverInfo.getLocation());
+            if (!isOpenShiftSupportEnabled(serverInfo)) {
+                serverInstanceKey = ModelFactory.newServerInstanceKey(id, serverInfo.getLocation());
                 serverTemplate.deleteServerInstance(serverInstanceKey.getServerInstanceId());
-
                 templateStorage.update(serverTemplate);
-
-                notifyOnDisconnect(serverInstanceKey, serverTemplate);
             }
+            logger.info("Server {} disconnected from controller", serverInstanceKey);
+
+            notifyOnDisconnect(serverInstanceKey, serverTemplate);
         }
     }
 
+    private boolean isOpenShiftSupportEnabled(KieServerInfo serverInfo) {
+        return serverInfo.getLocation() == null || serverInfo.getLocation().trim().length() == 0;
```

Comment:
Wouldn't it have more sense to check value of `org.kie.workbench.controller.openshift.enabled` instead of checking location?

Reply:
Re-implemented in that way is mainly due to package dependency constraints. Existing constant and method are defined at ```kie-wb-common``` [repo](https://github.com/rhtevan/kie-wb-common/blob/419cc606221f4c8408c86e977f91b98f9cf6c81f/kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-api/src/main/java/org/kie/workbench/common/screens/server/management/utils/ControllerUtils.java#L32). As the dependency chain is ```kie-wb-common``` --> ```droolsjbpm-integration```, referencing that constant from ```droolsjbpm-integration``` will introduce cycling dependency which should be avoided. Alternatively, it could be refactored out from ```kie-wb-common/ControllerUtils``` into ```droolsjbpm-integration/kie-server-controller-openshift```, but it would require renaming the property to something like: ```org.kie.controller.openshift.enabled```, that demands more coordinated changes impacting not only ```droolsjbpm-integration``` but also Business Automation Operator, BA-APB, RHPAM Templates, Documentation etc. Therefore, current implementation should be a pragmatic approach. 

@sutaakar Does it make sense? Any further suggestions?

Reply:
Sounds ok to me. Just please adjust the method name to something like isKieServerLocationAvailable() as we discussed on the call.

Reply:
@sutaakar Done the change. See if there were anything else, it should be good to merge.